### DISCUSSION TITLE
Not creating config files?

### DISCUSSION THREAD

**Discussion by untenops** [giuseppe99barchetta/SuggestArr#54] [Q&A]
I'm sure I'm doing something wrong here but I can't get this to create the config files.  Everything I try gets this error.

 - api_service.config.config - WARNING - /app/config/config_files/config.yaml not found. Creating a new one with default values.

 - api_service.blueprints.config.routes - ERROR - Error loading configuration: Object of type function is not JSON serializable

**Comment by giuseppe99barchetta** [comment]
> I'm sure I'm doing something wrong here but I can't get this to create the config files. Everything I try gets this error.
> 
> * api_service.config.config - WARNING - /app/config/config_files/config.yaml not found. Creating a new one with default values.
> * api_service.blueprints.config.routes - ERROR - Error loading configuration: Object of type function is not JSON serializable

Are you running it via Docker or locally with Python? If this is your first time running it, it's not actually an error; the system is scanning for existing configuration. If it doesn't find one, it returns an error. I need to handle this differently in the future.

**Reply by untenops** [reply]
Running with the docker compose file under "Docker Compose Example". 
   Sorry how do I get passed this?

**Reply by giuseppe99barchetta** [reply]
> Running with the docker compose file under "Docker Compose Example". 
> 
>    Sorry how do I get passed this?  

Where do you receive this error? And it's the first run?

**Reply by untenops** [reply]
Its the first run, using docker compose as yaml file.  
Using Portainer to watch for errors. 
![Screenshot 2024-10-25 103615](https://github.com/user-attachments/assets/39319936-88ff-40d5-ae47-4fadf11aff8f)
And also finding the errors in the GUI system logs.
![Screenshot 2024-10-25 103630](https://github.com/user-attachments/assets/1a061920-fd39-4b18-b3d0-78faf8cd6bcf)

**Reply by giuseppe99barchetta** [reply]
As I mentioned earlier, this is not a real error. When you open the web interface, it searches for a configuration file, which doesn’t exist yet. You can proceed; the file will be created later in the process.

**Reply by untenops** [reply]
Ok, but this is as far as it gets.  
![Screenshot 2024-10-25 104359](https://github.com/user-attachments/assets/ed9129ce-7348-4189-8953-111b594bed41)
And just stops here.

Looking for config files, a config_files directory is created but it is empty.

**Reply by giuseppe99barchetta** [reply]
> Ok, but this is as far as it gets. ![Screenshot 2024-10-25 104359](https://private-user-images.githubusercontent.com/4633453/380254062-ed9129ce-7348-4189-8953-111b594bed41.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk4NzUyNDAsIm5iZiI6MTcyOTg3NDk0MCwicGF0aCI6Ii80NjMzNDUzLzM4MDI1NDA2Mi1lZDkxMjljZS03MzQ4LTQxODktODk1My0xMTFiNTk0YmVkNDEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMjVUMTY0OTAwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MmUwNzhmYWE0NGVjNDkzOWMzZTVhNDA4ZTQ5ZjQwMjIwODljZjNkODFmNzE5YzMxYTc0M2JhZTMyMTA1NWRlZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.kJnqaWaS9RFI3gUk1V9sj5XoAjqB-JtZLVF2_SUtFro) And just stops here.
> 
> Looking for config files, a config_files directory is created but it is empty.

At this point, I recommend opening an issue for this. Please try creating an empty config.yaml file in that directory and see if that resolves your problem. If the issue persists, could you please share additional details about your system configuration and the settings for your container? This will help to better understand the problem.

**Reply by untenops** [reply]
adding the config.yaml  to the config_files directory changed the error to 

- api_service.blueprints.config.routes - ERROR - Error loading configuration: 'NoneType' object has no attribute 'get'

So no joy.
But Thanks for your time and effort.  If I get a chance later I'll look at opening a issue.
